### ISSUE TITLE
Install operator the Gitops way (with ArgoCD)

### ISSUE THREAD

**Issue by aelbarkani** [GoogleCloudPlatform/k8s-config-connector#432]
From the documentation it seems the only supported way of installing Config connector is through operator in non GKE clusters. The operator is deployed through a StatefulSet with this image: gcr.io/gke-release/cnrm/operator. That manifests doesn't seem to be nowhere in this repo, so it makes difficult to deploy the operator by using ArgoCD for example. Is there something we missed ? Is there somewhere (GCS or Github) where we can get plain non compressed yaml files to install the operator ?

**Comment by jcanseco** [comment]
Hi @aelbarkani, I'll admit I'm not all too familiar with ArgoCD, so please bear with me as I try to understand your use-case.

Are you asking if we could include the Operator's manifests in this GitHub repository so that you could have ArgoCD point to our repo to fetch the operator's manifests?

**Comment by aelbarkani** [comment]
Yes that's correct !

**Comment by jcanseco** [comment]
Hi @aelbarkani. Gotcha, thanks for the elaboration!

Yes we can definitely look into doing that for you. I just saw that you mentioned that GCS also works, so we'll look into uploading to GCS instead since that is simpler (and quicker) for us to implement given how our release pipeline works. Let us know if that won't be sufficient for you.

In the meantime, could you elaborate on the priority of this request? Are you blocked by any chance? (I'm assuming an interim workaround would be to host the operator's manifests on your own git repo while we update our release pipeline).

**Comment by aelbarkani** [comment]
Thanks @jcanseco ! I would say medium-low priority. We are not blocked since we are doing it manually, although a bit annoying if there are a lot of new releases.

**Comment by aelbarkani** [comment]
@jcanseco any update on this topic ?

**Comment by jcanseco** [comment]
Hey @aelbarkani, we haven't really been able to get to this, so we have no real updates yet. I'll bump the request internally for re-prioritization.

**Comment by jcanseco** [comment]
Hi @aelbarkani, so I just tried seeing if we can prioritize this request internally, but given our current priorities and capacity, I don't believe we will be able to work on this request for a while longer, unfortunately.

As a potential workaround: have you tried looking into the [GKE add-on](https://cloud.google.com/config-connector/docs/how-to/install-upgrade-uninstall)? The GKE add-on allows you to install Config Connector on GKE clusters without you having to apply the operator manifests yourselves. The add-on can also [automatically upgrade Config Connector](https://cloud.google.com/config-connector/docs/how-to/install-upgrade-uninstall#upgrading) for you whenever the GKE cluster is upgraded.

Please let us know if that works for you.

**Comment by procinger** [comment]
Hello, what is the situation here in the meantime?
I am currently supporting a team that wants to go full GitOps (ArgoCD) with GKE Autopilot Clusters.

Among other things, the Config Connector should spawn further GKE Autopilot clusters.

Unfortunately, Google does not offer any [Config Connector Addons for GKE Autopilot](https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/550). Therefore, we would also need the rendered manifest in a Git repository so that ArgoCD can synchronize the updates and rollout against the clusters.

I have just successfully executed this [setup](https://cloud.google.com/config-connector/docs/how-to/install-manually#installing_the_operator) in a GKE Autopilot cluster. Unfortunately, this is not possible in the ArgoCD context (download tarball and extract)

In ArgoCD, applications are defined as follows
https://argo-cd.readthedocs.io/en/stable/operator-manual/declarative-setup/#applications

**Comment by ldanielmadariaga** [comment]
This item has been open for >12 months. As part of scheduled repository maintenance, this Issue/PR will be closed. If this is still a critical ask, please re-open and work with your Google Cloud Account team to prioritize it with GCP KCC Product Management.
